module_macaddress.psm1

```diff
@@ -0,0 +1,102 @@
+    and hopefully retrieves the correct vendor of it. 
+
+.EXAMPLE
+    # Update Vendor List
+    Update-MacAddressVendor
+
+    # Search through vendor list
+    Get-MacAddressVendor -MacAddress '00-50-56-C0-00-00'
+
+    ===================RESULT===================
+    00:50:56	Vmware	VMware, Inc.
+    ============================================
+    
+
+.NOTES
+    Script written by CodeBarbarian
+#>
+
+
+
+# Incase of a proxy
+[System.Net.WebRequest]::DefaultWebProxy.Credentials = [System.Net.CredentialCache]::DefaultCredentials
+
+
+################################# Script Config ##################################
+# -------------------------- Script Dependant Variables -------------------------#
+# Document Path - Where the vendor mac addresses textfile will be located
+$DocumentPath = ("c:$($ENV:HOMEPATH)\desktop")
+##################################################################################
+
+
+
+function Update-MacAddressVendor {
+    # Path to the wireshark repository
+    $APIUrl = 'https://code.wireshark.org/review/gitweb?p=wireshark.git;a=blob_plain;f=manuf;hb=HEAD'
+
+    $WebClient = New-Object System.Net.WebClient
+    $WebClient.DownloadFile($APIUrl, "$($DocumentPath)\vendor_macaddresses.txt")
+
+    Write-Host ("Update of MAC vendor list complete.") -ForegroundColor Green
+}
+
+function Get-MacAddressVendor {
+    param (
+        [parameter(mandatory=$true)]
+        [string]$MacAddress
+    ) 
+
+    # Simple check for this one
+    if ([String]::IsNullOrWhiteSpace($MacAddress)) {
+        Write-Host ("The mac address can not be null or whitespace") -ForegroundColor RED
+    }
+
+    # Remove everything we really do not need from the mac address
+    $MacAddress = $MacAddress -replace "[-,.:]", ""
+    
+    # Prepare the placeholder for the search query
+    $Placeholder = "HOLD1:HOLD2:HOLD3"
+    $Placeholder = $Placeholder.Replace("HOLD1", $MacAddress.Substring(0,2))
+    $Placeholder = $Placeholder.Replace("HOLD2", $MacAddress.Substring(2,2))
+    $Placeholder = $Placeholder.Replace("HOLD3", $MacAddress.Substring(4,2))
+
+    # Set the mac address to become the mac address in same format as wireshark uses
+    $MacAddress = $Placeholder
+    
+    # Let us just make sure we did not fuck up somewhere
+    $RegEx = '(?<macaddr>[0-9A-Fa-f][0-9A-Fa-f]:[0-9A-Fa-f][0-9A-Fa-f]:[0-9A-Fa-f][0-9A-Fa-f])'
+
+    # Get contents from file
+    $File = Get-Content ("$($DocumentPath)\vendor_macaddresses.txt")
+
+    # Iterate through the file
+    foreach ($Line in $File) {
+        # Only check on Mac Address first
+        if ($Line -match $RegEx) {
+            # Check if the mac address is in the list
+            if ($Matches.values -Match $MacAddress) {
+                # Give response viewing the entire line
+
+                Write-Host("===================RESULT===================") -ForegroundColor White
+                Write-Host($Line) -ForegroundColor Green

```

Comment:
See comments above about `Write-Host`, i.e. please use `Write-Verbose` instead. Also a PowerShell command should return an object. Please add `return $Line`

Reply:
Sorry, I actually forgot to remove them. This was during testing earlier today, I usually just pipeline them directly using ""

Reply:
But yes, any function in powershell should return an object.

Reply:
No worries. You could also return is using just `$Line` if you are a purist but I have a slight preference for the return statement because it makes it easier to read.